### PR TITLE
New Spanish Cheat Sheet

### PR DESCRIPTION
**Pull request by rasikapohankar**
IA Page link: https://duck.co/ia/view/spanish_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@rasikapohankar Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by pnpninja** [comment]
@rasikapohankar : what is ur level of proficiency in spanish?

**Comment by rasikapohankar** [comment]
@pnpninja  : Level 0. I am still learning.

**Comment by pnpninja** [comment]
@rasikapohankar : this will take some time for reviewing by proficient speakers..

**Comment by pjhampton** [comment]
What is the difference between this cheat sheet and the cheat sheet here that @mayank7 opened? 

https://github.com/duckduckgo/zeroclickinfo-goodies/pull/2641

**Comment by mayank7** [comment]
@rasikapohankar I have already created this cheat sheet with lot more words and phrases, please check #2641

**Comment by rasikapohankar** [comment]
@pjhampton , @mayank7 : I created the Spanish Cheat Sheet IA page on 25th February but did not submit a pull request. The IA page link in #2641 points to the page created by me. But the pull request was submitted by @mayank7. So I submitted another pull request #2645 . What should I do?

**Comment by pjhampton** [comment]
I'll cc some community leaders in to get their opinion. 

cc/ @GuiltyDolphin, @mintsoft, @moollaza

**Comment by GuiltyDolphin** [comment]
@rasikapohankar @mayank7 @pjhampton The IA Page does indeed point to one created by @rasikapohankar - I think priority should be given to them on the PR.

@mayank7 @rasikapohankar Perhaps you would like to work together on this? Sharing ideas is never a bad thing, and you've both come up with a bunch of phrases!

@mayank7 For future reference - if someone has already made an IA page, it's best to check with them before beginning development; the usual case is that people make the IA pages *because* they want to make the IA.

@rasikapohankar I assume you want to continue development on this? If so leave this PR open and maybe @mayank7 will give you some tips on phrases.

@tagawa Want to see if anyone on Twitter wants to have a peer? I imagine we have quite a few fluent Spanish speakers!

**Comment by mintsoft** [comment]
OK, we also have 2 cheatsheets in this PR (regexpression) can this be removed from this branch @rasikapohankar

**Comment by rasikapohankar** [comment]
Yes I'll remove the regexpression cheat sheet. I had done that only to try
it out.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=spanish cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/spanish_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/language/spanish.json [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+            
+            {
+                "key": "¿De dónde eres?",
+                "val": "Where are you from?"
+            },
+            {
+                "key": "¿Cuántos años tienes?",
+                "val": "How old are you?"
+            },
+            {
```
Comment (gchaincl) [inline share/goodie/cheat_sheets/json/language/spanish.json]: This entry is duplicated

**Comment by tagawa** [comment]
@GuiltyDolphin Thanks - just tweeted: https://twitter.com/duckduckgo/status/704298850547269633

**Comment by dobladov** [comment]
These are the errors I found.

**Appears two times**

> ¿Cuántos años tienes?
> How old are you?

**Should be:** ¿Puedo ayudarle?
> ¿Podría ayudarse? 
> Can I help you? 

**Should be:** ¡Siga recto!, despúes, gire a la izquierda/derecha
> ¡Vaya ud derecho! Pues tuerza ud por la iquierda/ derecha! 
> Go straight! then turn left/ right

**Just need** "por favor" **at the end**
>  Puedo tener una servilleta?
> Can I have a napkin, please?

**Comment by tagawa** [comment]
Some of the questions have missing "¿" at the beginning, for example:

> Cuánto cuesta?

should be

> ¿Cuánto cuesta?

(Thanks to [feedback on Twitter](https://twitter.com/SrAparicio/status/704310420899811328))

**Comment by TeacherMary** [comment]
Spanish is my native language.
I found the following 
Solo Un Poquito should be Solo un poquito. 
Every single question must have the opening question mark ¿
Be careful with the overuse of exclamation marks. In Spanish might be taken differently. More as commands than as requests. For instance, ¡Venga conmigo! should be Venga conmigo. 
Arroz and alergias words have no accent marks Spanish. 
Regarding to the phrase "Tengo alergias." this might be taken as if you are having a bad reaction to something, say a food. If you want to say an excuse or an explanation why you won't have a certain food, you could say I'm allergic to...  Soy alérgico / alérgica a...

File: share/goodie/cheat_sheets/json/language/spanish.json [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+                "key": "Buenas noches",
+                "val": "Good evening"                
+            },
+            {
+                "key": "Buenas tardes",
+                "val": "Good afternoon"
+            },
+            {
+                "key": "Como estas?",
```
Comment (gchaincl) [inline share/goodie/cheat_sheets/json/language/spanish.json]: Missing tilde
- Como estás
+ Cómo estás

File: share/goodie/cheat_sheets/json/language/spanish.json [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+                "val": "See you later!"
+            }            		
+        ],
+        "Introductions": [
+            {
+                "key": "¿Hablas Inglés?",
+                "val": "Do you speak English?"
+            },
+            {
+                "key": "Solo Un Poquito.",
```
Comment (gchaincl) [inline share/goodie/cheat_sheets/json/language/spanish.json]: Missing tilde:
Sólo un poquito

File: share/goodie/cheat_sheets/json/language/spanish.json [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+            {
+                "key": "Estoy perdido",
+                "val": "I'm lost"
+            },
+            {
+                "key": "¿Podría ayudarse?",
+                "val": "Can I help you?"
+            },
+            {
+                "key": "¡Vaya ud derecho! Pues tuerza ud por la iquierda/ derecha!",
```
Comment (gchaincl) [inline share/goodie/cheat_sheets/json/language/spanish.json]: Wrong: Pues tuerza ud por la iquierda/ derecha!"
Good:Después doble por la izquierda/derecha

File: share/goodie/cheat_sheets/json/language/spanish.json [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+            {
+                "key": "Puedo tener agua, por favor?",
+                "val": "Can I have water,please?"
+            },
+            {
+                "key": "Una cerveza, por favor.",
+                "val": "One beer, please."
+            },
+            {
+                "key": "Puedo tener agua, por favor?",
```
Comment (gchaincl) [inline share/goodie/cheat_sheets/json/language/spanish.json]: This is duplicated.

File: share/goodie/cheat_sheets/json/language/spanish.json [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+            {
+                "key": "Tengo alergiás.",
+                "val": "I have allergies."
+            },
+            {
+                "key": "Soy vegetariano/vegan.",
+                "val": "I’m a vegetarian/vegan."
+            },
+            {
+                "key": "Me puede pasar la sal/el pimiento?",
```
Comment (gchaincl) [inline share/goodie/cheat_sheets/json/language/spanish.json]: wrong: el pimiento
good: la pimienta

**Comment by novia713** [comment]
there are some errors & typos besides the already commented.

- «_Tengo alergiás._» must be «_Tengo alergias._»
- «_¿Podría ayudarse?_» must be «_¿Podría/puede ayudarme?_»
- i'm spanish and in Spain, at least, we say «_tomar_» and not «_tener_» for «have» in «_can i have ..._»
so i would say «¿_Puedo tomar más, por favor?_» instead of «_¿Puedo tener mas?_»
there are several places where this error appears in the text.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=spanish cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/spanish_cheat_sheet_mobile.png)

**Comment by amoraleda** [comment]
Hi, I'm also native spanish speaker from Spain. 

In the particular case of `"Popote" / "Straw"` I would add that this is the mexican (?) word and that in Spain the proper word would be`"Paja"`.

And don't forget to add "¡" at the beginning of the exclamations or interjections.

----

**Sources:** [http://www.rae.es/](Real Academia Española)

**Comment by rasikapohankar** [comment]
@tagawa , @GuiltyDolphin , @mintsoft : Changed the template type to language to include pronunciations and added corrections from feedback. Can someone please also review the pronunciations?

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=spanish cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/spanish_cheat_sheet_mobile.png)

**Comment by GuiltyDolphin** [comment]
@rasikapohankar It looks like the `Eating Out` section is running down the page quite a lot:

screenshot from 2016-03-01 19 45 42

Maybe we could split this across a couple of sections? Or reduce the length a bit - we shouldn't have to scroll for one section.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=spanish cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/spanish_cheat_sheet_mobile.png)

**Comment by rasikapohankar** [comment]
@GuiltyDolphin : Yes the Eating Out section is indeed long. I'll reorganize the sections.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=spanish cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/spanish_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=spanish cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/spanish_cheat_sheet_mobile.png)

**Comment by GuiltyDolphin** [comment]
@rasikapohankar I'm still seeing the same issue, I think we need to actually _split_ the section up so that the space can be used more efficiently.

**Comment by rasikapohankar** [comment]
@GuiltyDolphin : Changed the order of sections to avoid the scrolling. Please check.

**Comment by GuiltyDolphin** [comment]
@rasikapohankar That looks better - weird, I checked a few minutes ago and the layout was still off!

:+1:

**Comment by rasikapohankar** [comment]
@GuiltyDolphin : I committed the changes just now. I think it was yet to be deployed when you checked.

**Comment by GuiltyDolphin** [comment]
@rasikapohankar Probably so! 

Have you made _all_ the changes suggested?

**Comment by rasikapohankar** [comment]
@GuiltyDolphin : Yes, removed the duplicates, added ¡ and ¿ where necessary, changed some of the words according to the suggestions. Should I ask everyone who gave suggestions earlier, if they still see anything wrong?

**Comment by GuiltyDolphin** [comment]
@rasikapohankar Great!

They'll probably be notified by the discussion - but if you want to see if anyone has any further suggestions feel free!

/cc @moollaza

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=spanish cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/spanish_cheat_sheet_mobile.png)

**Comment by moollaza** [comment]
Thanks! This LGTM :+1:

**Comment by rasikapohankar** [comment]
@moollaza: Thanks! :D

**Pull request merged**